pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -152,14 +200,18 @@ func buildClusterKey(service *model.Service, port *model.Port, cb *ClusterBuilde
 
 // buildOutboundClusters generates all outbound (including subsets) clusters for a given proxy.
 func (configgen *ConfigGeneratorImpl) buildOutboundClusters(cb *ClusterBuilder, cp clusterPatcher) ([]*discovery.Resource, cacheStats) {
-	resources := make([]*discovery.Resource, 0)
-	hit, miss := 0, 0
 	var services []*model.Service
 	if features.FilterGatewayClusterConfig && cb.proxy.Type == model.Router {
 		services = cb.push.GatewayServices(cb.proxy)
 	} else {
 		services = cb.push.Services(cb.proxy)
 	}
+	return configgen.buildOutboundClustersWithServices(cb, cp, services)
+}
+
+func (configgen *ConfigGeneratorImpl) buildOutboundClustersWithServices(cb *ClusterBuilder, cp clusterPatcher, services []*model.Service) ([]*discovery.Resource, cacheStats) {
```

Comment:
Where do you handle the deleted service?

Reply:
I see it in delta.go, but with this, i donot think it is right

Reply:
ping on this question

Reply:
i think i solved it

Reply:
ack